stdlib/map.mli

```diff
@@ -86,6 +86,16 @@ module type S =
        of [x] in [m] disappears.
        @before 4.03 Physical equality was not ensured. *)
 
+    val update: key -> ('a option -> 'a) -> 'a t -> 'a t
```

Comment:
Do you have a rationale for the signature of `f` ? My instinct would rather say `'a option -> 'a option`.